### PR TITLE
[Refactor RetinaNet] Step 8 Unify NMS Decoder

### PR DESCRIPTION
**Pull request by tanzhenyu**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: keras_cv/layers/object_detection/nms_prediction_decoder.py [hunk 1]
```diff
@@ -15,115 +15,98 @@
 import tensorflow as tf
 
 from keras_cv import bounding_box
-from keras_cv import layers as cv_layers
 
 
-class NmsPredictionDecoder(tf.keras.layers.Layer):
+# TODO(tanzhenyu): provide a TPU compatible NMS decoder.
```
Comment (bhack) [inline keras_cv/layers/object_detection/nms_prediction_decoder.py:20]: Can you add an expected failing test with jit_compile=True?

Comment (tanzhenyu) [reply keras_cv/layers/object_detection/nms_prediction_decoder.py:20]: What's expected failing test? Do you have an example?

**Comment by tanzhenyu** [comment]
closing this in favor of https://github.com/keras-team/keras-cv/pull/1197

File: keras_cv/layers/object_detection/nms_prediction_decoder.py (hunk 1 above)
Comment (bhack) [reply keras_cv/layers/object_detection/nms_prediction_decoder.py:20]: `@unittest.expectedFailure` with the NMSDecoder with `jit_compile=True`

**Pull request closed without merging**